core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1243,6 +1259,7 @@ private void completeFileInternal(RpcContext rpcContext, LockedInodePath inodePa
         .setId(inode.getId())
         .setUfsFingerprint(ufsFingerprint)
         .setLastModificationTimeMs(opTimeMs)
+        .setLastAccessTimeMs(opTimeMs)
```

Comment:
this is for create, but where do other accesses like mv get updated? Also, when you delete an inode, does the parent get updated to? I don't see that one either.

Reply:
rename and delete will only update the timestamps of the immediate parent directory. It is handled via `InodeTreePersistentState::updateTimestampsAndChildCount()`

Reply:
Rename on file system master eventually calls `InodeTreePersistentState::applyRename()` which calls `InodeTreePersistentState::updateTimestampsAndChildCount()` to update parent timestamps. It is considered a "content update" of parent directory thus should update both last modification time and last access time. This behavior is consistent with UNIX file system, while HDFS does not implement access time for directories.